doc/reorder.awk

```diff
@@ -0,0 +1,115 @@
+# This AWK script extracts typescripts comments from lsp-messages.ads
+# and reorders them in an order that corresponds to LSP specification.
+# Use next command to run the script:
+#
+#sed -n -e '/^   --```typescript/,/^   --```/p' \
+# ../source/protocol/lsp-messages.ads | awk -f reorder.awk
+BEGIN{Span_Index=0}
+/--```typescript/{Line_Index=0}
+!/--```/{ARR[Span_Index][Line_Index++]=$0}
+/--```$/{Span_Index++;Line_Index=0}
+END{
+    M=0
```

Comment:
this seems really hard to maintain - I'm not sure how it works... do we need to check this in?

Reply:
lsp-messages.ads contains typescript pieces in comments surrounded by kind of brackets:
```
--```typescript
--   ... typescript code here ...
--```
```
Typescript pieces are copied from LSP specification. The problem here is that we need to change their order, because Ada requires forward declarations, while LSP specification doesn't use this order. This script extracts typescripts pieces and prints them in original order. After that one can diff result with corresponding text from LSP specification.

I'm afraid to lost this information (relative order of typescript pieces), so until we have better way to do diff, I would prefer to keep this in repo.